zeek

```diff
@@ -89,15 +63,58 @@ start)
+		"$IMAGE_NAME" \
+		zeekcfg -o /zeek/node.cfg --type afpacket
+	fi
+
+	# create the volumes required for peristing user-installed zkg packages
+	$SUDO docker volume create zeek-zkg-script >/dev/null
+	$SUDO docker volume create zeek-zkg-plugin >/dev/null
+	$SUDO docker volume create zeek-zkg-state >/dev/null
+
+	docker_cmd=("docker" "run" "--detach")  # start container in the background
+	docker_cmd+=("--name" "$CONTAINER_NAME") # provide a predictable name
+	docker_cmd+=("--restart" "$RESTART_POLICY")
+	docker_cmd+=("--cap-add" "net_raw" "--cap-add" "net_admin") # allow Zeek to listen to raw packets
+	docker_cmd+=("--network" "host") # allow Zeek to monitor host network interfaces
+	# allow packages installed via zkg to persist across restarts
+	docker_cmd+=("--mount" "source=zeek-zkg-script,destination=/usr/local/zeek/share/zeek/site/packages/,type=volume")
+	docker_cmd+=("--mount" "source=zeek-zkg-plugin,destination=/usr/local/zeek/lib/zeek/plugins/packages/,type=volume")
+	docker_cmd+=("--mount" "source=zeek-zkg-state,destination=/root/.zkg,type=volume")
+	# mirror the host timezone settings to the container
+	docker_cmd+=("--mount" "source=/etc/localtime,destination=/etc/localtime,type=bind,readonly")
+	# persist and allow accessing the logs from the host
+	docker_cmd+=("--mount" "source=$HOST_ZEEK/logs,destination=/usr/local/zeek/logs/,type=bind")
+	docker_cmd+=("--mount" "source=$HOST_ZEEK/spool,destination=/usr/local/zeek/spool/,type=bind")
+	# allow users to provide arbitrary custom config files and scripts
+	# mount all zeekctl config files
+	while IFS=  read -r -d $'\0' CONFIG; do
+    	docker_cmd+=("--mount" "source=$CONFIG,destination=/usr/local/zeek/${CONFIG#"$HOST_ZEEK"},type=bind")
+	done < <(find "$HOST_ZEEK/etc/" -type f -print0 2>/dev/null)
+	# mount all zeek scripts
+	while IFS=  read -r -d $'\0' SCRIPT; do
+    	docker_cmd+=("--mount" "source=$SCRIPT,destination=/usr/local/zeek/${SCRIPT#"$HOST_ZEEK"},type=bind")
+	done < <(find "$HOST_ZEEK/share/" -type f -print0 2>/dev/null)
+		# loop reference: https://stackoverflow.com/a/23357277
+		# ${CONFIG#"$HOST_ZEEK"} strips $HOST_ZEEK prefix
+	docker_cmd+=("$IMAGE_NAME")
+
 	echo "Starting the Zeek docker container" >&2
-	$SUDO docker run --cap-add net_raw --cap-add net_admin --network host --detach \
-	 --name "$CONTAINER_NAME" \
-	 --restart always \
-	 --mount source=/etc/localtime,destination=/etc/localtime,type=bind,readonly \
-	 --mount "source=$host_zeek_logs,destination=/usr/local/zeek/logs/,type=bind" \
-	 --mount "source=$host_zeek_spool,destination=/usr/local/zeek/spool/,type=bind" \
-	 --mount "source=$host_zeek_node_cfg,destination=/usr/local/zeek/etc/node.cfg,type=bind" \
-	 "$IMAGE_NAME"
+	$SUDO "${docker_cmd[@]}"
+
+	# Fix current symlink for the host (sleep to give Zeek time to finish starting)
+	(sleep 15s; $SUDO docker exec "$CONTAINER_NAME" ln -sf "$HOST_ZEEK/spool/manager" /usr/local/zeek/logs/current) &
```

Comment:
Hat tip to @william-stearns for the idea here. This fixes the `logs/current` symlink so that it's valid on the host instead of inside the container. To my knowledge Zeek doesn't use this symlink internally and it's more useful to be valid on the host for the analyst. 

We have to sleep because Zeek takes time to start and it recreates this symlink every time it starts. If we do it right away Zeek will overwrite it.

I run the `ln` command in the container rather than on the host on the chance that the current user is part of the `docker` group but doesn't have write access to the host directory. This is pretty common in my personal experience so this quick privilege escalation makes it so I don't have to put in my sudo password every time I restart Zeek.